src/Aspire.Hosting.Keycloak/KeycloakResouceBuilderExtensions.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Keycloak;
+using Aspire.Hosting.Utils;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding Keycloak resources to an <see cref="IDistributedApplicationBuilder"/>.
+/// </summary>
+public static class KeycloakResouceBuilderExtensions
+{
+    private const string AdminEnvVarName = "KEYCLOAK_ADMIN";
+    private const string AdminPasswordEnvVarName = "KEYCLOAK_ADMIN_PASSWORD";
+    private const int DefaultContainerPort = 8080;
+
+    /// <summary>
+    /// Adds a Keycloak container to the application model.
+    /// </summary>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. </param>
+    /// <param name="port">The host port that the underlying container is bound to when running locally.</param>
+    /// <param name="admin">The parameter used as the admin for the Keycloak resource. If <see langword="null"/> a default value will be used.</param>
+    /// <param name="adminPassword">The parameter used as the admin password for the Keycloak resource. If <see langword="null"/> a default password will be used.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
```

Comment:
Please add more expansive `<remarks>` sections and an `<example>` section to each extension method here. What we want to enable with XML docs here is for people to see the reference docs on the MS Docs site virtually be able to just copy and paste some code into their own apphost.

Reply:
Added!